go/integrityblock/hash.go

```diff
@@ -0,0 +1,22 @@
+package integrityblock
+
+import (
+	"crypto/sha512"
+	"io"
+	"os"
+)
+
+// ComputeWebBundleSha512 computes the SHA-512 hash over the given web bundle file.
+func ComputeWebBundleSha512(bundleFile *os.File, offset int64) ([]byte, error) {
```

Comment:
```suggestion
func ComputeWebBundleSha512(bundleFile *io.ReadSeeker, offset int64) ([]byte, error) {
```
Then you can use non-file types that implement `io.ReadSeeker` (and still you can pass `*io.File` to this function). For example, `bytes.Reader` may be used in tests like this:

```go
bundleBytes = []byte{...}
got, err := ComputeWebBundleSha512(bytes.NewReader(bundleBytes), 0)
```